src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -178,4 +211,526 @@ export class NetworkAreaDiagramViewer {
+            endPolyline,
+            this.svgParameters.getArrowLabelShift()
+        );
+        const labelRotationElement = arrowElement.firstElementChild
+            ?.lastElementChild as SVGGraphicsElement;
+        labelRotationElement.setAttribute(
+            'transform',
+            'rotate(' + labelData[0].toFixed(2) + ')'
+        );
+        labelRotationElement.setAttribute('x', '' + labelData[1].toFixed(2));
+        if (labelData[2]) {
+            labelRotationElement.setAttribute('style', labelData[2]);
+        } else if (labelRotationElement.hasAttribute('style')) {
+            labelRotationElement.removeAttribute('style');
+        }
+    }
+
+    private moveTransformer(
+        edgeNode: SVGGraphicsElement,
+        startPolyline1: Point,
+        endPolyline1: Point,
+        startPolyline2: Point,
+        endPolyline2: Point,
+        edgeType: DiagramUtils.EdgeType
+    ) {
+        const transformerElement: SVGGraphicsElement =
+            edgeNode.lastElementChild as SVGGraphicsElement;
+        // move transformer circles
+        const transformerCircles: NodeListOf<SVGGraphicsElement> =
+            transformerElement?.querySelectorAll('circle');
+        this.moveTransformerCircle(
+            transformerCircles.item(0),
+            startPolyline1,
+            DiagramUtils.getDistance(
+                endPolyline1,
+                startPolyline1,
+                1.5 * this.svgParameters.getTransfomerCircleRadius()
+            )
+        );
+        this.moveTransformerCircle(
+            transformerCircles.item(1),
+            startPolyline2,
+            DiagramUtils.getDistance(
+                endPolyline2,
+                startPolyline2,
+                1.5 * this.svgParameters.getTransfomerCircleRadius()
+            )
+        );
+        // if phase shifting transformer move transformer arrow
+        const isPSTransformerEdge =
+            edgeType == DiagramUtils.EdgeType.PHASE_SHIFT_TRANSFORMER;
+        if (isPSTransformerEdge) {
+            this.moveTransformerArrow(
+                transformerElement,
+                startPolyline1,
+                endPolyline1,
+                DiagramUtils.getMidPosition(endPolyline1, endPolyline2)
+            );
+        }
+    }
+
+    private moveTransformerCircle(
+        transformerCircle: SVGGraphicsElement,
+        startPolyline: Point,
+        endPolyline: Point
+    ) {
+        const circleCenter: Point = DiagramUtils.getDistance(
+            endPolyline,
+            startPolyline,
+            -this.svgParameters.getTransfomerCircleRadius()
+        );
+        transformerCircle.setAttribute('cx', '' + circleCenter.x.toFixed(2));
+        transformerCircle.setAttribute('cy', '' + circleCenter.y.toFixed(2));
+    }
+
+    private moveTransformerArrow(
+        transformerElement: SVGGraphicsElement,
+        startPolyline: Point,
+        endPolyline: Point,
+        middle: Point
```

Comment:
Could you rename it into `transformerCenter`?

Reply:
done